kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -34,8 +35,41 @@
 
 public class LogSegment {
 
+  //////////////////////////////////////////
+  // Static factory methods and constants //
+  //////////////////////////////////////////
+
+  /**
+   * Creates a LogSegment from a single ParsedDeltaData. Used to construct a post-commit Snapshot
+   * after a CREATE transaction.
+   *
+   * @param logPath The path to the _delta_log directory
+   * @param parsedDelta The ParsedDeltaData (e.g., version 0)
+   * @return A new LogSegment with just this delta
+   * @throws IllegalArgumentException if the ParsedDeltaData is not file-based
+   */
+  public static LogSegment fromSingleDelta(Path logPath, ParsedDeltaData parsedDelta) {
```

Comment:
nit: are there other use-cases other then Create for the method (if so maybe name it more specifically?)